### PR TITLE
add types for missing ones and bring psalm-baseline down to zero

### PR DESCRIPTION
**Pull request by szaimen**
Signed-off-by: Simon L <szaimen@e.mail.de>

### CODE REVIEW

File: php/public/index.php [hunk 1]
```diff
@@ -12,6 +12,8 @@
 use Slim\Factory\AppFactory;
 use Slim\Views\Twig;
 use Slim\Views\TwigMiddleware;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\RequestInterface as Request;
```
Comment (szaimen) [inline php/public/index.php:16]: not completely sure if `ServerRequestInterface` would not be the better one and/or only for authentication handling... Would need to be tested.

Comment (szaimen) [reply php/public/index.php]: psalm added the solution

File: php/public/index.php [hunk 2]
```diff
@@ -140,7 +142,7 @@
 });
 
 // Auth Redirector
-$app->get('/', function (\Psr\Http\Message\RequestInterface $request, \Psr\Http\Message\ResponseInterface $response, $args) use ($container) {
+$app->get('/', function (\Psr\Http\Message\RequestInterface $request, Response $response, mixed $args) use ($container) {
```
Comment (come-nc) [inline php/public/index.php:145]:
```suggestion
$app->get('/', function (Request $request, Response $response, mixed $args) use ($container) {
```

consistency

**Review by come-nc — CHANGES_REQUESTED** [review]
I think `$args` should be `array`: https://www.slimframework.com/docs/v4/objects/routing.html#route-callbacks

File: php/public/index.php (hunk 2 above)
Comment (szaimen) [reply php/public/index.php]: No, RequestInterface should be correct IIRC... (It is not a ServerRequest but a Client-Side request IIRC)...

**Comment by szaimen** [comment]
> I think `$args` should be `array`: https://www.slimframework.com/docs/v4/objects/routing.html#route-callbacks

Thanks for figuring this out! Fixed :)

**Review by come-nc — APPROVED** [review]

File: php/public/index.php (hunk 2 above)
Comment (come-nc) [reply php/public/index.php]: Name is misleading then, maybe import both as Request and ServerRequest to avoid the confusion.

**Comment by szaimen** [comment]
Thanks a lot guys! :)

File: php/public/index.php (hunk 2 above)
Comment (szaimen) [reply php/public/index.php]: sorry, saw this too late

**Pull request merged**